ansible/roles/netplan/tasks/main.yml

```diff
@@ -0,0 +1,83 @@
+---
+# Copyright (C) 2015-2017 Maciej Delmanowski <drybjed@gmail.com>
+# Copyright (C) 2015-2017 Robin Schneider <ypid@riseup.net>
+# Copyright (C) 2015-2017 DebOps <https://debops.org/>
+# SPDX-License-Identifier: GPL-3.0-only
+
+# - name: Import DebOps global handlers
+#   ansible.builtin.import_role:
+#     name: 'global_handlers'
+
+# - name: Make sure that Ansible local facts directory exists
+#   ansible.builtin.file:
+#     path: '/etc/ansible/facts.d'
+#     state: 'directory'
+#     mode: '0755'
+
+# - name: Save netplan local facts
+#   ansible.builtin.template:
+#     src: 'etc/ansible/facts.d/netplan.fact.j2' # this file doesn't exist
+#     dest: '/etc/ansible/facts.d/netplan.fact'
+#     mode: '0755'
+#   notify: [ 'Refresh host facts' ]
+#   tags: [ 'meta::facts' ]
+
+# - name: Update Ansible facts if they have been modified
+#   ansible.builtin.meta: 'flush_handlers'
+
+- name: Install required packages
+  ansible.builtin.package:
+    name: '{{ q("flattened", (netplan__base_packages
+                              + netplan__packages)) }}'
+    state: 'present'
+  register: netplan__register_packages
+  until: netplan__register_packages is succeeded
+
+- name: Purge conflicting packages
+  ansible.builtin.apt:
+    name: '{{ q("flattened", netplan__purge_packages) }}'
+    state: 'absent'
+    purge: True
+
+- name: Purge existing netplan configs not deployed by this role
+  block:
+
+    - name: Locate existing netplan configurations
+      ansible.builtin.find:
+        paths: /etc/netplan
+        patterns: "*.yml,*.yaml"
+      register: _netplan_existing_configs
```

Comment:
In DebOps, the convention for runtime variables is `<role_name>__register_<variable>` so in this case it would be:
```yaml
register: netplan__register_existing_configs
```

Reply:
Remove the `_` prefix too. I know that it's a convention in Python to use them for private variables, but it's not needed in Ansible.